src/EventService.Validation/Event/CreateEventRequestValidator.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Linq;
+using FluentValidation;
+using LT.DigitalOffice.EventService.Broker.Requests.Interfaces;
+using LT.DigitalOffice.EventService.Data.Interfaces;
+using LT.DigitalOffice.EventService.Models.Dto.Enums;
+using LT.DigitalOffice.EventService.Models.Dto.Requests.Event;
+using LT.DigitalOffice.EventService.Validation.Event.Interfaces;
+using LT.DigitalOffice.Kernel.Extensions;
+using Microsoft.AspNetCore.Http;
+using Microsoft.IdentityModel.Tokens;
+
+namespace LT.DigitalOffice.EventService.Validation.Event;
+
+public class CreateEventRequestValidator : AbstractValidator<CreateEventRequest>, ICreateEventRequestValidator
+{
+  public CreateEventRequestValidator(
+    IUserService userService,
+    IHttpContextAccessor contextAccessor,
+    ICategoryRepository categoryRepository)
+  {
+    RuleFor(ev => ev.Name)
+      .MaximumLength(150)
+      .WithMessage("Name should not exceed maximum length of 150 symbols");
+
+    When(ev => !string.IsNullOrWhiteSpace(ev.Description), () =>
+    {
+      RuleFor(ev => ev.Description)
+        .MaximumLength(500)
+        .WithMessage("Description should not exceed maximum length of 500 symbols");
+    });
+
+    RuleFor(ev => ev.Address)
+      .MaximumLength(400)
+      .WithMessage("Address should not exceed maximum length of 400 symbols");
+
+    RuleFor(ev => ev.Date)
+      .Must(d => d > DateTime.UtcNow)
+      .WithMessage("The event date must be later than the date the event was created");
+
+    RuleFor(ev => ev.Users)
+      .Cascade(CascadeMode.Stop)
```

Comment:
this one is obsolete as VS says. It should be used like this:
![image](https://user-images.githubusercontent.com/101804724/235105841-b827d96b-f078-4f0b-aed7-0900e5ed833a.png)